Build error

```
In file included from /usr/include/boost/test/tools/floating_point_comparison.hpp:21:0,
                 from /usr/include/boost/test/tools/old/impl.hpp:21,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/boost/test/tools/detail/print_helper.hpp: In instantiation of ‘struct boost::test_tools::tt_detail::print_log_value<Color>’:
/usr/include/boost/test/tools/detail/print_helper.hpp:178:5:   required from ‘std::ostream& boost::test_tools::tt_detail::operator<<(std::ostream&, const boost::test_tools::tt_detail::print_helper_t<T>&) [with T = Color; std::ostream = std::basic_ostream<char>]’
/usr/include/boost/test/utils/lazy_ostream.hpp:66:29:   required from ‘std::ostream& boost::unit_test::lazy_ostream_impl<PrevType, T, StorageT>::operator()(std::ostream&) const [with PrevType = boost::unit_test::lazy_ostream; T = boost::test_tools::tt_detail::print_helper_t<Color>; StorageT = const boost::test_tools::tt_detail::print_helper_t<Color>&; std::ostream = std::basic_ostream<char>]’
/tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:59:24:   required from here
/usr/include/boost/test/tools/detail/print_helper.hpp:47:5: error: ‘value’ is not a member of ‘boost::has_left_shift<std::basic_ostream<char>, Color, boost::detail::has_left_shift_impl::dont_care>’
     BOOST_STATIC_ASSERT_MSG( (boost::has_left_shift<std::ostream,T>::value),
     ^
make[2]: *** [CMakeFiles/unitTests_parser.dir/build.make:63: CMakeFiles/unitTests_parser.dir/tests/parser/grammar-tests/color_grammar_test.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1315: CMakeFiles/unitTests_parser.dir/all] Error 2
make: *** [Makefile:139: all] Error 2
```

fuller log:

```
[ 28%] Built target server-obj
[ 36%] Built target alacarte-obj
[ 38%] Built target alacarte-maps-server
[ 40%] Built target alacarte-maps-importer
[ 48%] Built target alacarte-obj-test
[ 77%] Built target server-obj-test
[ 78%] Built target UnitTests-shared-obj
[ 78%] Building CXX object CMakeFiles/unitTests_parser.dir/tests/parser/grammar-tests/color_grammar_test.cpp.o
In file included from /usr/include/boost/iostreams/detail/is_dereferenceable.hpp:12:0,
                 from /usr/include/boost/iostreams/detail/resolve.hpp:26,
                 from /usr/include/boost/iostreams/detail/push.hpp:24,
                 from /usr/include/boost/iostreams/detail/streambuf/indirect_streambuf.hpp:31,
                 from /usr/include/boost/iostreams/stream_buffer.hpp:22,
                 from /usr/include/boost/iostreams/stream.hpp:21,
                 from /usr/include/boost/spirit/home/qi/stream/detail/iterator_source.hpp:14,
                 from /usr/include/boost/spirit/home/qi/stream/stream.hpp:16,
                 from /usr/include/boost/spirit/home/qi/stream.hpp:15,
                 from /usr/include/boost/spirit/home/qi.hpp:30,
                 from /usr/include/boost/spirit/include/qi.hpp:16,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:4:
/usr/include/boost/type_traits/detail/bool_trait_def.hpp:18:79: note: #pragma message: NOTE: Use of this header (bool_trait_def.hpp) is deprecated
 # pragma message("NOTE: Use of this header (bool_trait_def.hpp) is deprecated")
                                                                               ^
In file included from /usr/include/boost/type_traits/detail/bool_trait_def.hpp:21:0,
                 from /usr/include/boost/iostreams/detail/is_dereferenceable.hpp:12,
                 from /usr/include/boost/iostreams/detail/resolve.hpp:26,
                 from /usr/include/boost/iostreams/detail/push.hpp:24,
                 from /usr/include/boost/iostreams/detail/streambuf/indirect_streambuf.hpp:31,
                 from /usr/include/boost/iostreams/stream_buffer.hpp:22,
                 from /usr/include/boost/iostreams/stream.hpp:21,
                 from /usr/include/boost/spirit/home/qi/stream/detail/iterator_source.hpp:14,
                 from /usr/include/boost/spirit/home/qi/stream/stream.hpp:16,
                 from /usr/include/boost/spirit/home/qi/stream.hpp:15,
                 from /usr/include/boost/spirit/home/qi.hpp:30,
                 from /usr/include/boost/spirit/include/qi.hpp:16,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:4:
/usr/include/boost/type_traits/detail/template_arity_spec.hpp:13:84: note: #pragma message: NOTE: Use of this header (template_arity_spec.hpp) is deprecated
 # pragma message("NOTE: Use of this header (template_arity_spec.hpp) is deprecated")
                                                                                    ^
In file included from /usr/include/boost/iostreams/detail/is_dereferenceable.hpp:13:0,
                 from /usr/include/boost/iostreams/detail/resolve.hpp:26,
                 from /usr/include/boost/iostreams/detail/push.hpp:24,
                 from /usr/include/boost/iostreams/detail/streambuf/indirect_streambuf.hpp:31,
                 from /usr/include/boost/iostreams/stream_buffer.hpp:22,
                 from /usr/include/boost/iostreams/stream.hpp:21,
                 from /usr/include/boost/spirit/home/qi/stream/detail/iterator_source.hpp:14,
                 from /usr/include/boost/spirit/home/qi/stream/stream.hpp:16,
                 from /usr/include/boost/spirit/home/qi/stream.hpp:15,
                 from /usr/include/boost/spirit/home/qi.hpp:30,
                 from /usr/include/boost/spirit/include/qi.hpp:16,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:4:
/usr/include/boost/type_traits/detail/template_arity_spec.hpp:13:84: note: #pragma message: NOTE: Use of this header (template_arity_spec.hpp) is deprecated
 # pragma message("NOTE: Use of this header (template_arity_spec.hpp) is deprecated")
                                                                                    ^
In file included from /usr/include/boost/config.hpp:61:0,
                 from /usr/include/boost/test/test_tools.hpp:17,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/boost/type_traits/detail/has_binary_operator.hpp: In instantiation of ‘const bool boost::detail::has_left_shift_impl::operator_exists<std::basic_ostream<char>, Color>::value’:
/usr/include/boost/type_traits/detail/has_binary_operator.hpp:179:4:   required from ‘const bool boost::detail::has_left_shift_impl::trait_impl1<std::basic_ostream<char>, Color, boost::detail::has_left_shift_impl::dont_care, false>::value’
/usr/include/boost/type_traits/detail/has_binary_operator.hpp:208:4:   required from ‘const bool boost::detail::has_left_shift_impl::trait_impl<std::basic_ostream<char>, Color, boost::detail::has_left_shift_impl::dont_care>::value’
/usr/include/boost/type_traits/detail/has_binary_operator.hpp:216:8:   required from ‘struct boost::has_left_shift<std::basic_ostream<char>, Color, boost::detail::has_left_shift_impl::dont_care>’
/usr/include/boost/test/tools/detail/print_helper.hpp:47:5:   required from ‘struct boost::test_tools::tt_detail::print_log_value<Color>’
/usr/include/boost/test/tools/detail/print_helper.hpp:178:5:   required from ‘std::ostream& boost::test_tools::tt_detail::operator<<(std::ostream&, const boost::test_tools::tt_detail::print_helper_t<T>&) [with T = Color; std::ostream = std::basic_ostream<char>]’
/usr/include/boost/test/utils/lazy_ostream.hpp:66:29:   required from ‘std::ostream& boost::unit_test::lazy_ostream_impl<PrevType, T, StorageT>::operator()(std::ostream&) const [with PrevType = boost::unit_test::lazy_ostream; T = boost::test_tools::tt_detail::print_helper_t<Color>; StorageT = const boost::test_tools::tt_detail::print_helper_t<Color>&; std::ostream = std::basic_ostream<char>]’
/tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:59:24:   required from here
/usr/include/boost/type_traits/detail/has_binary_operator.hpp:158:70: error: ambiguous overload for ‘operator< ’ (operand types are ‘std::basic_ostream<char>’ and ‘Color’)
    BOOST_STATIC_CONSTANT(bool, value = (sizeof(s_check(((make<Lhs>() BOOST_TT_TRAIT_OP make<Rhs>()),make<has_operator>())))==sizeof(::boost::type_traits::yes_type)));
                                                                      ^
In file included from /usr/include/c++/6.2.1/iterator:64:0,
                 from /usr/include/boost/test/detail/workaround.hpp:19,
                 from /usr/include/boost/test/detail/global_typedef.hpp:16,
                 from /usr/include/boost/test/tree/observer.hpp:17,
                 from /usr/include/boost/test/unit_test_log.hpp:18,
                 from /usr/include/boost/test/tools/old/impl.hpp:19,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/c++/6.2.1/ostream:166:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(long int) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(long __n)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:170:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(long unsigned int) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(unsigned long __n)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:174:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(bool) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(bool __n)
       ^~~~~~~~
In file included from /usr/include/c++/6.2.1/ostream:638:0,
                 from /usr/include/c++/6.2.1/iterator:64,
                 from /usr/include/boost/test/detail/workaround.hpp:19,
                 from /usr/include/boost/test/detail/global_typedef.hpp:16,
                 from /usr/include/boost/test/tree/observer.hpp:17,
                 from /usr/include/boost/test/unit_test_log.hpp:18,
                 from /usr/include/boost/test/tools/old/impl.hpp:19,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/c++/6.2.1/bits/ostream.tcc:91:5: note: candidate: std::basic_ostream<_CharT, _Traits>& std::basic_ostream<_CharT, _Traits>::operator<<(short int) [with _CharT = char; _Traits = std::char_traits<char>]
     basic_ostream<_CharT, _Traits>::
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/6.2.1/iterator:64:0,
                 from /usr/include/boost/test/detail/workaround.hpp:19,
                 from /usr/include/boost/test/detail/global_typedef.hpp:16,
                 from /usr/include/boost/test/tree/observer.hpp:17,
                 from /usr/include/boost/test/unit_test_log.hpp:18,
                 from /usr/include/boost/test/tools/old/impl.hpp:19,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/c++/6.2.1/ostream:181:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(short unsigned int) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(unsigned short __n)
       ^~~~~~~~
In file included from /usr/include/c++/6.2.1/ostream:638:0,
                 from /usr/include/c++/6.2.1/iterator:64,
                 from /usr/include/boost/test/detail/workaround.hpp:19,
                 from /usr/include/boost/test/detail/global_typedef.hpp:16,
                 from /usr/include/boost/test/tree/observer.hpp:17,
                 from /usr/include/boost/test/unit_test_log.hpp:18,
                 from /usr/include/boost/test/tools/old/impl.hpp:19,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/c++/6.2.1/bits/ostream.tcc:105:5: note: candidate: std::basic_ostream<_CharT, _Traits>& std::basic_ostream<_CharT, _Traits>::operator<<(int) [with _CharT = char; _Traits = std::char_traits<char>]
     basic_ostream<_CharT, _Traits>::
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/6.2.1/iterator:64:0,
                 from /usr/include/boost/test/detail/workaround.hpp:19,
                 from /usr/include/boost/test/detail/global_typedef.hpp:16,
                 from /usr/include/boost/test/tree/observer.hpp:17,
                 from /usr/include/boost/test/unit_test_log.hpp:18,
                 from /usr/include/boost/test/tools/old/impl.hpp:19,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/c++/6.2.1/ostream:192:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(unsigned int) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(unsigned int __n)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:201:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(long long int) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(long long __n)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:205:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(long long unsigned int) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(unsigned long long __n)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:220:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(double) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(double __f)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:224:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(float) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(float __f)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:232:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(long double) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(long double __f)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:245:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(const void*) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(const void* __p)
       ^~~~~~~~
In file included from /usr/include/boost/type_traits/has_left_shift.hpp:43:0,
                 from /usr/include/boost/test/tools/detail/print_helper.hpp:29,
                 from /usr/include/boost/test/tools/old/impl.hpp:24,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/boost/type_traits/detail/has_binary_operator.hpp:70:13: note: candidate: boost::detail::has_left_shift_impl::no_operator boost::detail::has_left_shift_impl::operator<<(const boost::detail::has_left_shift_impl::any&, const boost::detail::has_left_shift_impl::any&)
 no_operator operator BOOST_TT_TRAIT_OP (const any&, const any&);
             ^~~~~~~~
In file included from /usr/include/c++/6.2.1/iterator:64:0,
                 from /usr/include/boost/test/detail/workaround.hpp:19,
                 from /usr/include/boost/test/detail/global_typedef.hpp:16,
                 from /usr/include/boost/test/tree/observer.hpp:17,
                 from /usr/include/boost/test/unit_test_log.hpp:18,
                 from /usr/include/boost/test/tools/old/impl.hpp:19,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/c++/6.2.1/ostream:519:5: note: candidate: std::basic_ostream<char, _Traits>& std::operator<<(std::basic_ostream<char, _Traits>&, unsigned char) [with _Traits = std::char_traits<char>]
     operator<<(basic_ostream<char, _Traits>& __out, unsigned char __c)
     ^~~~~~~~
/usr/include/c++/6.2.1/ostream:514:5: note: candidate: std::basic_ostream<char, _Traits>& std::operator<<(std::basic_ostream<char, _Traits>&, signed char) [with _Traits = std::char_traits<char>]
     operator<<(basic_ostream<char, _Traits>& __out, signed char __c)
     ^~~~~~~~
/usr/include/c++/6.2.1/ostream:508:5: note: candidate: std::basic_ostream<char, _Traits>& std::operator<<(std::basic_ostream<char, _Traits>&, char) [with _Traits = std::char_traits<char>]
     operator<<(basic_ostream<char, _Traits>& __out, char __c)
     ^~~~~~~~
/usr/include/c++/6.2.1/ostream:502:5: note: candidate: std::basic_ostream<_CharT, _Traits>& std::operator<<(std::basic_ostream<_CharT, _Traits>&, char) [with _CharT = char; _Traits = std::char_traits<char>]
     operator<<(basic_ostream<_CharT, _Traits>& __out, char __c)
     ^~~~~~~~
In file included from /usr/include/boost/config.hpp:61:0,
                 from /usr/include/boost/test/test_tools.hpp:17,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/boost/type_traits/detail/has_binary_operator.hpp: In instantiation of ‘const bool boost::detail::has_left_shift_impl::operator_returns_void<std::basic_ostream<char>, Color>::value’:
/usr/include/boost/type_traits/detail/has_binary_operator.hpp:179:4:   required from ‘const bool boost::detail::has_left_shift_impl::trait_impl1<std::basic_ostream<char>, Color, boost::detail::has_left_shift_impl::dont_care, false>::value’
/usr/include/boost/type_traits/detail/has_binary_operator.hpp:208:4:   required from ‘const bool boost::detail::has_left_shift_impl::trait_impl<std::basic_ostream<char>, Color, boost::detail::has_left_shift_impl::dont_care>::value’
/usr/include/boost/type_traits/detail/has_binary_operator.hpp:216:8:   required from ‘struct boost::has_left_shift<std::basic_ostream<char>, Color, boost::detail::has_left_shift_impl::dont_care>’
/usr/include/boost/test/tools/detail/print_helper.hpp:47:5:   required from ‘struct boost::test_tools::tt_detail::print_log_value<Color>’
/usr/include/boost/test/tools/detail/print_helper.hpp:178:5:   required from ‘std::ostream& boost::test_tools::tt_detail::operator<<(std::ostream&, const boost::test_tools::tt_detail::print_helper_t<T>&) [with T = Color; std::ostream = std::basic_ostream<char>]’
/usr/include/boost/test/utils/lazy_ostream.hpp:66:29:   required from ‘std::ostream& boost::unit_test::lazy_ostream_impl<PrevType, T, StorageT>::operator()(std::ostream&) const [with PrevType = boost::unit_test::lazy_ostream; T = boost::test_tools::tt_detail::print_helper_t<Color>; StorageT = const boost::test_tools::tt_detail::print_helper_t<Color>&; std::ostream = std::basic_ostream<char>]’
/tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:59:24:   required from here
/usr/include/boost/type_traits/detail/has_binary_operator.hpp:95:114: error: ambiguous overload for ‘operator< ’ (operand types are ‘std::basic_ostream<char>’ and ‘Color’)
 C_CONSTANT(bool, value = (sizeof(::boost::type_traits::yes_type)==sizeof(returns_void((make<Lhs>() BOOST_TT_TRAIT_OP make<Rhs>(),returns_void_t())))));
                                                                                                    ^
In file included from /usr/include/c++/6.2.1/iterator:64:0,
                 from /usr/include/boost/test/detail/workaround.hpp:19,
                 from /usr/include/boost/test/detail/global_typedef.hpp:16,
                 from /usr/include/boost/test/tree/observer.hpp:17,
                 from /usr/include/boost/test/unit_test_log.hpp:18,
                 from /usr/include/boost/test/tools/old/impl.hpp:19,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/c++/6.2.1/ostream:166:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(long int) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(long __n)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:170:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(long unsigned int) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(unsigned long __n)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:174:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(bool) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(bool __n)
       ^~~~~~~~
In file included from /usr/include/c++/6.2.1/ostream:638:0,
                 from /usr/include/c++/6.2.1/iterator:64,
                 from /usr/include/boost/test/detail/workaround.hpp:19,
                 from /usr/include/boost/test/detail/global_typedef.hpp:16,
                 from /usr/include/boost/test/tree/observer.hpp:17,
                 from /usr/include/boost/test/unit_test_log.hpp:18,
                 from /usr/include/boost/test/tools/old/impl.hpp:19,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/c++/6.2.1/bits/ostream.tcc:91:5: note: candidate: std::basic_ostream<_CharT, _Traits>& std::basic_ostream<_CharT, _Traits>::operator<<(short int) [with _CharT = char; _Traits = std::char_traits<char>]
     basic_ostream<_CharT, _Traits>::
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/6.2.1/iterator:64:0,
                 from /usr/include/boost/test/detail/workaround.hpp:19,
                 from /usr/include/boost/test/detail/global_typedef.hpp:16,
                 from /usr/include/boost/test/tree/observer.hpp:17,
                 from /usr/include/boost/test/unit_test_log.hpp:18,
                 from /usr/include/boost/test/tools/old/impl.hpp:19,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/c++/6.2.1/ostream:181:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(short unsigned int) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(unsigned short __n)
       ^~~~~~~~
In file included from /usr/include/c++/6.2.1/ostream:638:0,
                 from /usr/include/c++/6.2.1/iterator:64,
                 from /usr/include/boost/test/detail/workaround.hpp:19,
                 from /usr/include/boost/test/detail/global_typedef.hpp:16,
                 from /usr/include/boost/test/tree/observer.hpp:17,
                 from /usr/include/boost/test/unit_test_log.hpp:18,
                 from /usr/include/boost/test/tools/old/impl.hpp:19,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/c++/6.2.1/bits/ostream.tcc:105:5: note: candidate: std::basic_ostream<_CharT, _Traits>& std::basic_ostream<_CharT, _Traits>::operator<<(int) [with _CharT = char; _Traits = std::char_traits<char>]
     basic_ostream<_CharT, _Traits>::
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/6.2.1/iterator:64:0,
                 from /usr/include/boost/test/detail/workaround.hpp:19,
                 from /usr/include/boost/test/detail/global_typedef.hpp:16,
                 from /usr/include/boost/test/tree/observer.hpp:17,
                 from /usr/include/boost/test/unit_test_log.hpp:18,
                 from /usr/include/boost/test/tools/old/impl.hpp:19,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/c++/6.2.1/ostream:192:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(unsigned int) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(unsigned int __n)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:201:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(long long int) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(long long __n)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:205:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(long long unsigned int) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(unsigned long long __n)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:220:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(double) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(double __f)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:224:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(float) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(float __f)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:232:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(long double) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(long double __f)
       ^~~~~~~~
/usr/include/c++/6.2.1/ostream:245:7: note: candidate: std::basic_ostream<_CharT, _Traits>::__ostream_type& std::basic_ostream<_CharT, _Traits>::operator<<(const void*) [with _CharT = char; _Traits = std::char_traits<char>; std::basic_ostream<_CharT, _Traits>::__ostream_type = std::basic_ostream<char>]
       operator<<(const void* __p)
       ^~~~~~~~
In file included from /usr/include/boost/type_traits/has_left_shift.hpp:43:0,
                 from /usr/include/boost/test/tools/detail/print_helper.hpp:29,
                 from /usr/include/boost/test/tools/old/impl.hpp:24,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/boost/type_traits/detail/has_binary_operator.hpp:70:13: note: candidate: boost::detail::has_left_shift_impl::no_operator boost::detail::has_left_shift_impl::operator<<(const boost::detail::has_left_shift_impl::any&, const boost::detail::has_left_shift_impl::any&)
 no_operator operator BOOST_TT_TRAIT_OP (const any&, const any&);
             ^~~~~~~~
In file included from /usr/include/c++/6.2.1/iterator:64:0,
                 from /usr/include/boost/test/detail/workaround.hpp:19,
                 from /usr/include/boost/test/detail/global_typedef.hpp:16,
                 from /usr/include/boost/test/tree/observer.hpp:17,
                 from /usr/include/boost/test/unit_test_log.hpp:18,
                 from /usr/include/boost/test/tools/old/impl.hpp:19,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/c++/6.2.1/ostream:519:5: note: candidate: std::basic_ostream<char, _Traits>& std::operator<<(std::basic_ostream<char, _Traits>&, unsigned char) [with _Traits = std::char_traits<char>]
     operator<<(basic_ostream<char, _Traits>& __out, unsigned char __c)
     ^~~~~~~~
/usr/include/c++/6.2.1/ostream:514:5: note: candidate: std::basic_ostream<char, _Traits>& std::operator<<(std::basic_ostream<char, _Traits>&, signed char) [with _Traits = std::char_traits<char>]
     operator<<(basic_ostream<char, _Traits>& __out, signed char __c)
     ^~~~~~~~
/usr/include/c++/6.2.1/ostream:508:5: note: candidate: std::basic_ostream<char, _Traits>& std::operator<<(std::basic_ostream<char, _Traits>&, char) [with _Traits = std::char_traits<char>]
     operator<<(basic_ostream<char, _Traits>& __out, char __c)
     ^~~~~~~~
/usr/include/c++/6.2.1/ostream:502:5: note: candidate: std::basic_ostream<_CharT, _Traits>& std::operator<<(std::basic_ostream<_CharT, _Traits>&, char) [with _CharT = char; _Traits = std::char_traits<char>]
     operator<<(basic_ostream<_CharT, _Traits>& __out, char __c)
     ^~~~~~~~
In file included from /usr/include/boost/test/tools/floating_point_comparison.hpp:21:0,
                 from /usr/include/boost/test/tools/old/impl.hpp:21,
                 from /usr/include/boost/test/test_tools.hpp:46,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/../../tests.hpp:15,
                 from /tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:2:
/usr/include/boost/test/tools/detail/print_helper.hpp: In instantiation of ‘struct boost::test_tools::tt_detail::print_log_value<Color>’:
/usr/include/boost/test/tools/detail/print_helper.hpp:178:5:   required from ‘std::ostream& boost::test_tools::tt_detail::operator<<(std::ostream&, const boost::test_tools::tt_detail::print_helper_t<T>&) [with T = Color; std::ostream = std::basic_ostream<char>]’
/usr/include/boost/test/utils/lazy_ostream.hpp:66:29:   required from ‘std::ostream& boost::unit_test::lazy_ostream_impl<PrevType, T, StorageT>::operator()(std::ostream&) const [with PrevType = boost::unit_test::lazy_ostream; T = boost::test_tools::tt_detail::print_helper_t<Color>; StorageT = const boost::test_tools::tt_detail::print_helper_t<Color>&; std::ostream = std::basic_ostream<char>]’
/tmp/16.23.21/alacarte/tests/parser/grammar-tests/color_grammar_test.cpp:59:24:   required from here
/usr/include/boost/test/tools/detail/print_helper.hpp:47:5: error: ‘value’ is not a member of ‘boost::has_left_shift<std::basic_ostream<char>, Color, boost::detail::has_left_shift_impl::dont_care>’
     BOOST_STATIC_ASSERT_MSG( (boost::has_left_shift<std::ostream,T>::value),
     ^
make[2]: *** [CMakeFiles/unitTests_parser.dir/build.make:63: CMakeFiles/unitTests_parser.dir/tests/parser/grammar-tests/color_grammar_test.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1315: CMakeFiles/unitTests_parser.dir/all] Error 2
make: *** [Makefile:139: all] Error 2
```

--- a/include/general/configuration.hpp
+++ b/include/general/configuration.hpp
@@ -115,8 +115,7 @@ namespace opt {
 class Configuration : public boost::enable_shared_from_this<Configuration>
 {
 public:
-	Configuration();
-
+	Configuration() = default;
 	Configuration(	boost::program_options::options_description& cmd_desc,
 					boost::program_options::options_description& config_desc,
 					boost::program_options::positional_options_description& pos_desc,

--- a/include/general/geo_object.hpp
+++ b/include/general/geo_object.hpp
@@ -35,9 +35,12 @@ class GeoObject
 private:
 	friend class boost::serialization::access;
 public:
-	GeoObject();
+	GeoObject() = default;
+	GeoObject(const GeoObject& other) = default;
+	GeoObject(GeoObject&& other) = default;
 	GeoObject(const DataMap<CachedString, CachedString>& tags);
-	
+	virtual ~GeoObject() = default;
+
 	//! Returns a map with key-to-tag-mapping for osm-tags. 
 	TESTABLE const DataMap<CachedString, CachedString>& getTags() const;
 protected:

--- a/include/general/geodata.hpp
+++ b/include/general/geodata.hpp
@@ -41,8 +41,8 @@ class RTree;
 class Geodata
 {
 public:
-	Geodata();
-	~Geodata();
+	Geodata() = default;
+	virtual ~Geodata() = default;
 
 	TESTABLE void insertNodes(const shared_ptr<std::vector<Node> >& nodes);
 	TESTABLE void insertWays(const shared_ptr<std::vector<Way> >& ways);

--- a/include/general/node.hpp
+++ b/include/general/node.hpp
@@ -30,11 +30,14 @@ class Node : public GeoObject
 private:
 	friend class boost::serialization::access;
 public:
+	Node() = default;
+	Node(const Node& other) = default;
+	Node(Node&& other) = default;
 	Node(const FloatPoint& location, const DataMap<CachedString, CachedString>& tags);
+	virtual ~Node() = default;
 
 	TESTABLE const FixedPoint& getLocation() const;
 private:
-	Node();
 	template<typename Archive>
 	void serialize(Archive &ar, const unsigned int version) {
 		GeoObject::serialize(ar, version);

--- a/include/general/relation.hpp
+++ b/include/general/relation.hpp
@@ -33,11 +33,15 @@ class Relation : public GeoObject
 private:
 	friend class boost::serialization::access;
 public:
+	Relation() = default;
+	Relation(const Relation& other) = default;
+	Relation(Relation&& other) = default;
 	Relation(	const std::vector<NodeId>& nodeIDs, 
 				const DataMap<NodeId, CachedString>& nodeRoles,
 				const std::vector<WayId>& wayIDs,
 				const DataMap<WayId, CachedString>& wayRoles,
 				const DataMap<CachedString, CachedString>& tags);
+	virtual ~Relation() = default;
 	
 	TESTABLE const std::vector<WayId>& getWayIDs() const;
 	TESTABLE const std::vector<NodeId>& getNodeIDs() const;    
@@ -53,7 +57,6 @@ private:
 		ar & wayRoles;
 	}
 
-	Relation();
 private:
 	std::vector<NodeId> nodeIDs;
 	std::vector<WayId> wayIDs;

--- a/include/general/rtree.hpp
+++ b/include/general/rtree.hpp
@@ -100,7 +100,7 @@ private:
 	};
 
 public:
-	RTree () {}
+	RTree() = default;
 	void build(std::vector<data_t>& data, const string& leafPath);
 	bool search ( boost::shared_ptr<std::vector<id_t> >& result,
 				  const FixedRect& rect, bool returnOnFirst = false ) const;

--- a/include/general/way.hpp
+++ b/include/general/way.hpp
@@ -33,11 +33,14 @@ class Way : public GeoObject
 private:
     friend class boost::serialization::access;
 public:
-
+	Way() = default;
+	Way(const Way& other) = default;
+	Way(Way&& other) = default;
 	Way(const std::vector<NodeId>& nodeIDs, const DataMap<CachedString, CachedString>& tags)
 	: GeoObject(tags)
 	, nodeIDs(nodeIDs)
 	{ }
+	virtual ~Way() = default;
 
 	inline const std::vector<NodeId>& getNodeIDs() const { return nodeIDs; }
 	inline bool isClosed() const { return (nodeIDs.front() == nodeIDs.back()); }
@@ -48,7 +51,6 @@ private:
 		GeoObject::serialize(ar, version);
 		ar & nodeIDs;
 	}
-	Way(){}
 
 private:
 	std::vector<NodeId> nodeIDs;

--- a/include/server/render_attributes.hpp
+++ b/include/server/render_attributes.hpp
@@ -38,10 +38,6 @@ class RenderAttributes
 public:
 	static const unsigned int ChunkSize = 1024;
 
-	/**
-	 * @brief Creates a new, empty RenderAttributes.
-	 */
-	RenderAttributes() {}
 	TESTABLE inline const boost::unordered_map<WayId, Style*>& getWayMap() const
 	{
 		return wayStyles;
@@ -62,8 +58,6 @@ public:
 	TESTABLE Style* getNewStyle(NodeId id)	{ return newStyle(id, nodeStyles); }
 	TESTABLE Style* getNewStyle(WayId id)	{ return newStyle(id, wayStyles); }
 	TESTABLE Style* getNewStyle(RelId id)	{ return newStyle(id, relationStyles); }
-
-
 
 private:
 	template<typename IdType>

--- a/include/server/renderer/render_canvas.hpp
+++ b/include/server/renderer/render_canvas.hpp
@@ -73,7 +73,7 @@ public:
 		LAYER_NUM
 	};
 
-	virtual ~RenderCanvas() {}
+	virtual ~RenderCanvas() = default;
 
 	//! Clears every layer. Should always be called before use.
 	virtual void clear() = 0;

--- a/include/utils/color.hpp
+++ b/include/utils/color.hpp
@@ -75,23 +75,28 @@ public:
 	inline Color& operator /=	(const Color& c)	{r /= c.r; g /= c.g; b /= c.b; a /= c.a; return *this;}
 	inline Color& operator /=	(const float f)		{r /= f; g /= f; b /= f; a /= f; return *this;}
 
-	inline Color operator + (const Color& _c)	{return Color(r + _c.r, g + _c.g, b + _c.b, a + _c.a);}
-	inline Color operator - (const Color& _c)	{return Color(r - _c.r, g - _c.g, b - _c.b, a - _c.a);}
-	inline Color operator - ()					{return Color(-r, -g, -b, a);}
-	inline Color operator * (const Color& _c)	{return Color(r * _c.r, g * _c.g, b * _c.b, a * _c.a);}
-	inline Color operator * (const float f)		{return Color(r * f, g * f, b * f, a * f);}
-	inline Color operator / (const Color& _c)	{return Color(r / _c.r, g / _c.g, b / _c.b, a / _c.a);}
-	inline Color operator / (const float f)		{return Color(r / f, g / f, b / f, a / f);}
+	inline Color operator + (const Color& _c) const	{return Color(r + _c.r, g + _c.g, b + _c.b, a + _c.a);}
+	inline Color operator - (const Color& _c) const	{return Color(r - _c.r, g - _c.g, b - _c.b, a - _c.a);}
+	inline Color operator - () const 					{return Color(-r, -g, -b, a);}
+	inline Color operator * (const Color& _c) const	{return Color(r * _c.r, g * _c.g, b * _c.b, a * _c.a);}
+	inline Color operator * (const float f) const		{return Color(r * f, g * f, b * f, a * f);}
+	inline Color operator / (const Color& _c) const	{return Color(r / _c.r, g / _c.g, b / _c.b, a / _c.a);}
+	inline Color operator / (const float f) const		{return Color(r / f, g / f, b / f, a / f);}
 
-	inline bool operator == (const Color& _c) {return (uint32)*this == (uint32)_c;}
-	inline bool operator != (const Color& _c) {return (uint32)*this != (uint32)_c;}
+	inline bool operator == (const Color& _c) const {return (uint32)*this == (uint32)_c;}
+	inline bool operator != (const Color& _c) const {return (uint32)*this != (uint32)_c;}
 
 
-	inline Color	negate()		{ return Color(1.0f - r, 1.0f - g, 1.0f - b, a); }
-	inline Color	negateA()		{ return Color(1.0f - r, 1.0f - g, 1.0f - b, 1.0f - a); }
-	inline Color	min(const Color& _c)	{ return Color(std::min(r, _c.r), std::min(g, _c.g), std::min(b, _c.b), std::min(a, _c.a)); }
-	inline Color	max(const Color& _c)	{ return Color(std::max(r, _c.r), std::max(g, _c.g), std::max(b, _c.b), std::max(a, _c.a)); }
+	inline Color	negate()  const		{ return Color(1.0f - r, 1.0f - g, 1.0f - b, a); }
+	inline Color	negateA() const		{ return Color(1.0f - r, 1.0f - g, 1.0f - b, 1.0f - a); }
+	inline Color	min(const Color& _c) const	{ return Color(std::min(r, _c.r), std::min(g, _c.g), std::min(b, _c.b), std::min(a, _c.a)); }
+	inline Color	max(const Color& _c) const	{ return Color(std::max(r, _c.r), std::max(g, _c.g), std::max(b, _c.b), std::max(a, _c.a)); }
 
+	friend std::ostream& operator<< (std::ostream& stream, const Color& color)
+	{
+		stream << "Color(" << color.r << "," << color.g << "," << color.b << "," << color.a << ")";
+		return stream;
+	}
 
 	//inline Color&	rand()			{ r = Random::Rand<float>(1.0f); greturn (*this); }
 

--- a/include/utils/color.hpp
+++ b/include/utils/color.hpp
@@ -40,15 +40,30 @@
 class Color
 {
 public:
-	Color()	: r(1.0f), g(1.0f), b(1.0f), a(1.0f)	{}
-	Color(const Color& _c) : r(_c.r), g(_c.g), b(_c.b), a(_c.a)																																						{}
-	Color(const float _f) : r(_f), g(_f), b(_f), a(1.0f)																																							{}
-	explicit Color(const float _r, const float _g, const float _b) : r(_r), g(_g), b(_b), a(1.0f)																													{}
-	explicit Color(const float _r, const float _g, const float _b, const float _a) : r(_r), g(_g), b(_b), a(_a)																										{}
-	explicit Color(uint8 _r, uint8 _g, uint8 _b) : r((float)(_r) * cxCOLOR_CONV), g((float)(_g) * cxCOLOR_CONV), b((float)(_b) * cxCOLOR_CONV), a(1.0f)																{}
-	explicit Color(uint8 _r, uint8 _g, uint8 _b, uint8 _a) : r((float)(_r) * cxCOLOR_CONV), g((float)(_g) * cxCOLOR_CONV), b((float)(_b) * cxCOLOR_CONV), a((float)(_a) * cxCOLOR_CONV)								{}
-	Color(const float* _c) : r(_c[0]), g(_c[1]), b(_c[2]), a(_c[3])																																					{}
-	Color(const uint8* pComponent) : r((float)(pComponent[0]) * cxCOLOR_CONV), g((float)(pComponent[1]) * cxCOLOR_CONV), b((float)(pComponent[2]) * cxCOLOR_CONV), a((float)(pComponent[3]) * cxCOLOR_CONV)			{}
+	Color() : r(1.0f), g(1.0f), b(1.0f), a(1.0f) {}
+	Color(const Color& _c) : r(_c.r), g(_c.g), b(_c.b), a(_c.a) {}
+	Color(const float _f) : r(_f), g(_f), b(_f), a(1.0f) {}
+	explicit Color(const float _r, const float _g, const float _b)
+		: r(_r), g(_g), b(_b), a(1.0f) {}
+	explicit Color(const float _r, const float _g, const float _b, const float _a)
+		: r(_r), g(_g), b(_b), a(_a) {}
+	explicit Color(uint8 _r, uint8 _g, uint8 _b)
+		: r((float)(_r) * cxCOLOR_CONV)
+		, g((float)(_g) * cxCOLOR_CONV)
+		, b((float)(_b) * cxCOLOR_CONV)
+		, a(1.0f) {}
+	explicit Color(uint8 _r, uint8 _g, uint8 _b, uint8 _a)
+		: r((float)(_r) * cxCOLOR_CONV)
+		, g((float)(_g) * cxCOLOR_CONV)
+		, b((float)(_b) * cxCOLOR_CONV)
+		, a((float)(_a) * cxCOLOR_CONV) {}
+	Color(const float* _c)
+		: r(_c[0]), g(_c[1]), b(_c[2]), a(_c[3]) {}
+	Color(const uint8* pComponent)
+		: r((float)(pComponent[0]) * cxCOLOR_CONV)
+		, g((float)(pComponent[1]) * cxCOLOR_CONV)
+		, b((float)(pComponent[2]) * cxCOLOR_CONV)
+		, a((float)(pComponent[3]) * cxCOLOR_CONV) {}
 
 	Color(uint32 _c)
 		: r(cxCOLOR_CONV * (float)(uint8)(_c >> 16))
@@ -87,10 +102,10 @@ public:
 	inline bool operator != (const Color& _c) const {return (uint32)*this != (uint32)_c;}
 
 
-	inline Color	negate()  const		{ return Color(1.0f - r, 1.0f - g, 1.0f - b, a); }
-	inline Color	negateA() const		{ return Color(1.0f - r, 1.0f - g, 1.0f - b, 1.0f - a); }
-	inline Color	min(const Color& _c) const	{ return Color(std::min(r, _c.r), std::min(g, _c.g), std::min(b, _c.b), std::min(a, _c.a)); }
-	inline Color	max(const Color& _c) const	{ return Color(std::max(r, _c.r), std::max(g, _c.g), std::max(b, _c.b), std::max(a, _c.a)); }
+	inline Color negate()             const { return Color(1.0f - r, 1.0f - g, 1.0f - b, a); }
+	inline Color negateA()            const { return Color(1.0f - r, 1.0f - g, 1.0f - b, 1.0f - a); }
+	inline Color min(const Color& _c) const { return Color(std::min(r, _c.r), std::min(g, _c.g), std::min(b, _c.b), std::min(a, _c.a)); }
+	inline Color max(const Color& _c) const { return Color(std::max(r, _c.r), std::max(g, _c.g), std::max(b, _c.b), std::max(a, _c.a)); }
 
 	friend std::ostream& operator<< (std::ostream& stream, const Color& color)
 	{

--- a/include/utils/point.hpp
+++ b/include/utils/point.hpp
@@ -39,16 +39,18 @@ private:
 		ar & y;
 	}
 public:
-	basic_vector2()	: x(T(0)), y(T(0))									{}
-	basic_vector2(const basic_vector2& _v)	: x(_v.x), y(_v.y)			{}
-	basic_vector2(const T _v)				: x(_v), y(_v)				{}
-	basic_vector2(const T _x, const T _y)	: x(_x), y(_y)				{}
+	basic_vector2()	: x(T(0)), y(T(0)) {}
+	basic_vector2(const basic_vector2& _v) = default;
+	basic_vector2(basic_vector2&& _v) = default;
+	basic_vector2(const T _v)             : x(_v), y(_v) {}
+	basic_vector2(const T _x, const T _y) : x(_x), y(_y) {}
 
 	// Casting-Operatoren
 	operator float* () {return (float*)(c);}
 
 	// Zuweisungsoperatoren
-	basic_vector2& operator = (const basic_vector2& v)	{x = v.x; y = v.y;		return (*this);}
+	basic_vector2& operator = (const basic_vector2& v) = default;
+	basic_vector2& operator = (basic_vector2&& v) = default;
 	basic_vector2& operator += (const basic_vector2& v)	{x += v.x; y += v.y;	return (*this);}
 	basic_vector2& operator -= (const basic_vector2& v)	{x -= v.x; y -= v.y;	return (*this);}
 	basic_vector2& operator *= (const basic_vector2& v)	{x *= v.x; y *= v.y;	return (*this);}

--- a/include/utils/statistic.hpp
+++ b/include/utils/statistic.hpp
@@ -52,7 +52,7 @@ public:
 	class JobMeasurement {
 		friend class Statistic;
 	public:
-		JobMeasurement(){};
+		JobMeasurement() = default;
 		JobMeasurement(const string& stylesheet, int zoom)
 			: stylesheet(stylesheet), zoom(zoom), nodes(0), ways(0), relations(0)
 		{

--- a/src/alacarte_server.cpp
+++ b/src/alacarte_server.cpp
@@ -180,7 +180,7 @@ protected:
 
 
 	/**
-	 * @brief Contains the main functionality for the server                                                                  
+	 * @brief Contains the main functionality for the server
 	 *
 	 * @param config the configuration for the application
 	 **/

--- a/src/general/configuration.cpp
+++ b/src/general/configuration.cpp
@@ -26,16 +26,6 @@
 
 
 /**
- * @brief Creats an empty configuration
- *
- **/
-Configuration::Configuration()
-{
-
-}
-
-
-/**
  * @brief Creates a new config and parses the command line and config file.
  *
  * Parses the command line arguments and stores them internally.

--- a/src/general/geo_object.cpp
+++ b/src/general/geo_object.cpp
@@ -27,8 +27,6 @@ GeoObject::GeoObject(const DataMap<CachedString, CachedString>& tags)
 {
 }
 
-GeoObject::GeoObject() {}
-
 const DataMap<CachedString, CachedString>& GeoObject::getTags() const
 {
 	return tags;

--- a/src/general/geodata.cpp
+++ b/src/general/geodata.cpp
@@ -45,14 +45,6 @@
 #include "utils/rect.hpp"
 #include "utils/archive.hpp"
 
-Geodata::Geodata()
-{
-}
-
-Geodata::~Geodata()
-{
-}
-
 //! called when data is serialized to file
 void Geodata::buildTrees(const string& nodePath, const string& wayPath, const string& relationPath)
 {

--- a/src/general/node.cpp
+++ b/src/general/node.cpp
@@ -38,7 +38,6 @@ Node::Node(const FloatPoint& sphereLocation, const DataMap<CachedString, CachedS
 {
 	projectMercator(sphereLocation, location.x, location.y);
 }
-Node::Node() {}
 
 
 const FixedPoint& Node::getLocation() const

--- a/src/general/relation.cpp
+++ b/src/general/relation.cpp
@@ -36,8 +36,6 @@ Relation::Relation(	const std::vector<NodeId>& nodeIDs,
 {
 }
 
-Relation::Relation() {}
-
 const std::vector<WayId>& Relation::getWayIDs() const
 {
 	return wayIDs;

--- a/src/server/renderer/render_canvas.cpp
+++ b/src/server/renderer/render_canvas.cpp
@@ -59,7 +59,7 @@ public:
 
 	PNGRenderCanvas(unsigned int layerWidth, unsigned int layerHeight,
 					unsigned int sliceWidth, unsigned int sliceHeight);
-	virtual ~PNGRenderCanvas() {}
+	virtual ~PNGRenderCanvas() = default;
 	virtual void clear();
 	virtual CairoLayer* getImageLayers() { return layers; }
 	virtual CairoLayer& getSliceLayer() { return slice; }

--- a/tests/general/geodata-test/geodataMock.cpp
+++ b/tests/general/geodata-test/geodataMock.cpp
@@ -20,12 +20,8 @@
 
 
 
-GeodataMock::GeodataMock()
-{
-}
-GeodataMock::GeodataMock(const Geodata& data) : Geodata(data) 
-{}
-GeodataMock::~GeodataMock()
+GeodataMock::GeodataMock(const Geodata& data)
+	: Geodata(data)
 {
 }
 

--- a/tests/general/geodata-test/geodataMock.cpp
+++ b/tests/general/geodata-test/geodataMock.cpp
@@ -36,46 +36,44 @@ void GeodataMock::insertWays(const shared_ptr<std::vector<Way> >& ways)
 }
 
 void GeodataMock::insertRelations(const shared_ptr<std::vector<Relation> >& relations)
-{	
-		this->relations = relations;
+{
+	this->relations = relations;
 }
-
-
 
 shared_ptr<std::vector<NodeId> > GeodataMock::getNodeIDs(const FixedRect& rect) const
 {
 	shared_ptr<std::vector<NodeId> > nodeIDs = boost::make_shared< std::vector<NodeId> >();
- 	for(int i = 0; i < nodes->size(); i++) {
- 		if(rect.contains(nodes->at(i).getLocation())) {
- 			nodeIDs->push_back(NodeId(i));
- 		}
- 	}
+	for(int i = 0; i < nodes->size(); i++) {
+		if(rect.contains(nodes->at(i).getLocation())) {
+			nodeIDs->push_back(NodeId(i));
+		}
+	}
 	return nodeIDs;
 }
 
- shared_ptr<std::vector<WayId> > GeodataMock::getWayIDs(const FixedRect& rect) const
- {
- 	shared_ptr<std::vector<WayId> > wayIDs = boost::make_shared< std::vector<WayId> >();
- 	for(int i = 0; i < ways->size(); i++) {
- 		if(rect.intersects(calculateBoundingBox(&ways->at(i)))) {
- 				wayIDs->push_back(WayId(i));
- 		}
- 	}
- 	return wayIDs;
- }
+shared_ptr<std::vector<WayId> > GeodataMock::getWayIDs(const FixedRect& rect) const
+{
+	shared_ptr<std::vector<WayId> > wayIDs = boost::make_shared< std::vector<WayId> >();
+	for(int i = 0; i < ways->size(); i++) {
+		if(rect.intersects(calculateBoundingBox(&ways->at(i)))) {
+				wayIDs->push_back(WayId(i));
+		}
+	}
+	return wayIDs;
+}
 
 
- shared_ptr<std::vector<RelId> > GeodataMock::getRelationIDs(const FixedRect& rect) const
- {
+shared_ptr<std::vector<RelId> > GeodataMock::getRelationIDs(const FixedRect& rect) const
+{
 	shared_ptr<std::vector<RelId> > relationIDs = boost::make_shared< std::vector<RelId> >();
 
- 	for(int i = 0; i < relations->size(); i++) {
- 		if ( rect.intersects ( calculateBoundingBox(&relations->at(i)))) {
+	for(int i = 0; i < relations->size(); i++) {
+		if ( rect.intersects ( calculateBoundingBox(&relations->at(i)))) {
 			relationIDs->push_back(RelId(i));
- 		}
- 	}
- 	return relationIDs;
- }
+		}
+	}
+	return relationIDs;
+}
 
 Node* GeodataMock::getNode(NodeId id) const
 {
@@ -88,14 +86,13 @@ Way* GeodataMock::getWay(WayId id) const
 }
 
 
-FixedRect GeodataMock::calculateBoundingBox(const Way* way) const {
-
+FixedRect GeodataMock::calculateBoundingBox(const Way* way) const
+{
 	return calculateBoundingBox(way->getNodeIDs());
 }
 
-FixedRect GeodataMock::calculateBoundingBox(const Relation* relation) const {
-
-
+FixedRect GeodataMock::calculateBoundingBox(const Relation* relation) const
+{
 	coord_t maxX = std::numeric_limits<coord_t>::min();
 	coord_t maxY = std::numeric_limits<coord_t>::min();
 	coord_t minX = std::numeric_limits<coord_t>::max();
@@ -126,8 +123,8 @@ FixedRect GeodataMock::calculateBoundingBox(const Relation* relation) const {
 	return FixedRect(minX, minY, maxX, maxY);
 }
 
-FixedRect GeodataMock::calculateBoundingBox(const std::vector<NodeId>& nodeIDs) const {
-
+FixedRect GeodataMock::calculateBoundingBox(const std::vector<NodeId>& nodeIDs) const
+{
 	FixedPoint help;
 	coord_t maxX = std::numeric_limits<coord_t>::min();
 	coord_t maxY = std::numeric_limits<coord_t>::min();

--- a/tests/general/geodata-test/geodataMock.hpp
+++ b/tests/general/geodata-test/geodataMock.hpp
@@ -36,12 +36,12 @@ class Way;
 class Relation;
 class Geodata;
 
-class GeodataMock : public Geodata						//erbt von Geodata
+class GeodataMock : public Geodata
 {
 public:
-	GeodataMock();
+	GeodataMock() = default;
 	GeodataMock(const Geodata& data);
-	~GeodataMock();
+	virtual ~GeodataMock() = default;
 
 	TESTABLE void insertNodes(const shared_ptr<std::vector<Node> >& nodes);
 	TESTABLE void insertWays(const shared_ptr<std::vector<Way> >& ways);

--- a/tests/mapcss/mocks/mock_geodata.cpp
+++ b/tests/mapcss/mocks/mock_geodata.cpp
@@ -7,8 +7,6 @@
 
 class MockGeodata : public Geodata {
 public:
-	MockGeodata() : Geodata() {}
-
 	MOCK_CONST_METHOD1(getNode, Node*(NodeId));
 	MOCK_CONST_METHOD1(getWay, Way*(WayId));
 	MOCK_CONST_METHOD1(getRelation, Relation*(RelId));

--- a/tests/mapcss/mocks/mock_selector.cpp
+++ b/tests/mapcss/mocks/mock_selector.cpp
@@ -8,7 +8,7 @@
 class MockSelector : public Selector {
 public:
 	MockSelector(shared_ptr<Rule> rule) : Selector(rule, shared_ptr<Selector>()) {}
-	virtual ~MockSelector() {}
+	virtual ~MockSelector() = default;
 
 	MOCK_CONST_METHOD3(matchNode, void(NodeId nodeID, const shared_ptr<TileIdentifier>& ti, RenderAttributes* attributes));
 	MOCK_CONST_METHOD3(matchWay, void(WayId wayID, const shared_ptr<TileIdentifier>& ti, RenderAttributes* attributes));

--- a/tests/server/non_prerendering_request_manager.hpp
+++ b/tests/server/non_prerendering_request_manager.hpp
@@ -18,7 +18,7 @@ public:
 	{
 	}
 	
-	virtual ~NonPrerenderingRequestManager() {}
+	virtual ~NonPrerenderingRequestManager() = default;
 	//Disabled to prevent prerendering
 	void enqueue(const shared_ptr<TileIdentifier>& ti)
 	{

